students/Sahlberg/Lesson5/mailroom3/mailroom3.py

```diff
@@ -0,0 +1,112 @@
+    return user_action
+
+def email(name, amount):
+    """Create email for specific donor and amount donated."""
+    return 'Thank you {} for the generous donation of $ {:.2f}. We appreciate your generosity.\n\nSincerely, \n\nThe Helping R Us Team\n\n'.format(name.title(), amount)
+
+def quit():
+    """Quit the program"""
+    return print('Until next time!')
+
+def sort_and_stats_dict(dictionary):
+    """Create a stats dictionary from a donor dictionary and sort by total donated."""
+    pre_sort = {k: [sum(v), len(v), round((sum(v) / len(v)), 0)] for k, v in dictionary.items()}
+    print(pre_sort)
+    donor_sorted = sorted(list(pre_sort.items()), key=lambda x: x[1], reverse=True)
+    return donor_sorted
+
+def print_report(dictionary):
+    """Print out a report of donors and amounts donated, including stats (total donated, number of times donated, average donation)."""
+    print('Donor Name       |      Donor Total    |  Count of Donations    |  Average of Donations')
+    print('----------------------------------------------------------------------------------------')
+    count = 0
+    for donor in dictionary:
+        print(
+            f'{dictionary[count][0]:<20}$    {dictionary[count][1][0]:<23.2f}{dictionary[count][1][1]:<15}    ${dictionary[count][1][2]:>12.0f}')
+        count += 1
+
+def email_file(dictionary):
+    """Generates a thank you letter and writes to disk for all donors."""
+    count = 0
+    count_file = 0
+    for name in dictionary.keys():
+        amount = [sum(v) for v in dictionary.values()][count]
+        file_name = '{}_{}.txt'.format(name, dt.date.today())
+        count += 1
+        with open(file_name, 'w+') as file:
+            e_text = str(email(name, amount))
+            count_file += 1
+            file.write(e_text)
+    return print('Emails written to file.')
+
+#---------------------------------------------------------------
+#Work Zone
+if __name__== '__main__':
+    user_action = user_options()
+
+    if user_action.lower() == 'quit':
+        quit()
+
+    while user_action == '1':
+        option = input(user_select.get(4))
+
+        if option.lower() == 'quit':
+            print('Bye!')
+            break
+
+        if option.lower() == 'list':
+            print_list(donor)
+        else:
+            if option.lower() in donor:
+                amount = donate(option,donor)
+                amount_append(option,donor,amount)
+                email(option,amount)
+
+            else:
+                donor[option] = []
+                amount = donate(option,donor)
+                amount_append(option, donor, amount)
+                email(option,amount)
+
+    while user_action == '2':
+        donor_sorted = sort_and_stats_dict(donor)
+        print_report(donor_sorted)
+        break
+
+    while user_action == '3':
+        email_file(donor)
```

Comment:
you should be using a switch dict pattern here (from Lesson 4) with exception handling.